tests/integration_tests/test_mlflow.py

```diff
@@ -0,0 +1,130 @@
+import py
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import optuna
+from optuna.integration.mlflow import MLflowCallback
+from tests.test_structs import _create_frozen_trial
+from tests.test_study import func
```

Comment:
I don’t think we should reuse those functions, because they seems not to be designed for usages outside `test_structs` and `test_study`. If we need similar logic, there would be two alternative options:
- (a) defining similar functions in `test_mlflow`, or
- (b) extracting the functions to more common places to somewhere under `optuna.testing`.

I personally much prefer (a), because I think `optuna.testing` should be the place for more common utilities than those functions. 

Even though we'd make some small duplicated logic, how about just taking option (a)?

Reply:
Ok. I refactored the functions into `test_mlflow`.